packages/js/src/ai-optimizer/components/ai-optimize-button.js

```diff
@@ -70,28 +72,62 @@ const AIOptimizeButton = ( { id, isPremium = false } ) => {
 	// (4) all blocks are in visual mode.
 	// eslint-disable-next-line complexity
 	const { isEnabled, ariaLabel } = useSelect( ( select ) => {
-		if ( activeAIButtonId !== null && ! isButtonPressed ) {
+		// When Premium add-on is not active (upsell), always show the generic tooltip
+		if ( shouldShowUpsell ) {
+			// Keep enablement tied to visual mode state, but always show "Optimize with AI" as tooltip
+			if ( editorType === "blockEditor" ) {
+				const blocks = getAllBlocks( select( "core/block-editor" ).getBlocks() );
+				const allVisual = editorMode === "visual" && blocks.every( block => select( "core/block-editor" ).getBlockMode( block.clientId ) === "visual" );
+				return {
+					isEnabled: allVisual,
+					ariaLabel: defaultLabel,
+				};
+			}
+			// Classic editor
 			return {
-				isEnabled: false,
-				ariaLabel: null,
+				isEnabled: editorMode === "visual",
+				ariaLabel: defaultLabel,
 			};
 		}
+		const keyphraseAssessments = [ "introductionKeyword", "keyphraseDensity", "keyphraseDistribution" ];
+		if ( keyphraseAssessments.includes( id ) ) {
+			const hasValidKeyphrase = !! keyword && keyword.trim().length > 0;
+			const collectData = get( window, "YoastSEO.analysis.collectData", false );
+			// Ensures the button uses the same analysis-ready content source, while staying safe if the analysis API hasn’t initialized.
+			const editorData = collectData ? collectData() : {};
+			const text = editorData?._text || "";
+			const hasContent = text.trim().length > 0;
+
+			// Check global disabled reasons first (for unsupported content)
+			const disabledAIButtons = select( "yoast-seo/editor" ).getDisabledAIFixesButtons();
+			if ( Object.keys( disabledAIButtons ).includes( aiOptimizeId ) ) {
+				// Show global disabled reason only if there is some content (unsupported content case)
+				if ( hasContent ) {
+					return {
+						isEnabled: false,
+						ariaLabel: disabledAIButtons[ aiOptimizeId ],
+					};
+				}
+			}
 
-		const disabledAIButtons = select( "yoast-seo/editor" ).getDisabledAIFixesButtons();
-		if ( Object.keys( disabledAIButtons ).includes( aiOptimizeId ) ) {
-			return {
-				isEnabled: false,
-				ariaLabel: disabledAIButtons[ aiOptimizeId ],
-			};
+			// If missing keyphrase or missing content, ask user to add both
+			if ( ! hasValidKeyphrase || ! hasContent ) {
+				return {
+					isEnabled: false,
+					ariaLabel: __( "Please add both a keyphrase and some text to your content.", "wordpress-seo" ),
+				};
+			}
 		}
 
+		// Editor mode check
```

Comment:
I found out that with this approach, the global disabled reason is always shown regardless of the editor mode. I wonder if this is always the case and if this is expected. 🤔